File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/studyroom/StudyRoomsPagerAdapter.java
package de.tum.in.tumcampusapp.component.ui.studyroom;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import de.tum.in.tumcampusapp.utils.Const;

/**
 * A {@link FragmentStatePagerAdapter} that returns a fragment corresponding to one
 * of the sections/tabs/pages.
 */
public class StudyRoomsPagerAdapter extends FragmentStatePagerAdapter {

    private int mCurrentStudyRoomGroupId;

    StudyRoomsPagerAdapter(FragmentManager fm) {
        super(fm);
    }

    /**
     * getItem is called to instantiate the fragment for the given page.
     *
     * @param position spinner position
     * @return the fragment
     */
    @Override
    public Fragment getItem(int position) {
        Fragment fragment = new StudyRoomGroupDetailsFragment();
        Bundle args = new Bundle();
        args.putInt(Const.STUDY_ROOM_GROUP_ID, mCurrentStudyRoomGroupId);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public int getCount() {
        return 1;
    }

    void setStudyRoomGroupId(int selectedGroupId) {
        mCurrentStudyRoomGroupId = selectedGroupId;
    }
}

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/studyroom/StudyRoomsPagerAdapter.kt
package de.tum.`in`.tumcampusapp.component.ui.studyroom

import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter

import de.tum.`in`.tumcampusapp.utils.Const

class StudyRoomsPagerAdapter(fm: FragmentManager) : FragmentStatePagerAdapter(fm) {

    private var currentId: Int = 0

    /**
     * Create a StudyRoomGroupDetailsFragment with the correct parameter for the selected group
     */
    override fun getItem(position: Int): Fragment {
        return StudyRoomGroupDetailsFragment().apply {
            arguments = Bundle().apply {
                putInt(Const.STUDY_ROOM_GROUP_ID, currentId)
            }
        }
    }

    override fun getCount() = 1

    internal fun setStudyRoomGroupId(selectedGroupId: Int) {
        currentId = selectedGroupId
    }
}
